### ISSUE TITLE
Track differences between Engine documentation and Moho

### ISSUE THREAD

**Issue by Hdt80bro** [FAForever/fa#4045]
Classes
-----
The engine documentation for classes is pretty good right now-- if it's in `/engine/Sim*` it's generally sim-side only, if it's in `/engine/User*`, user-side, and `/engine/Core*` for both. It's not quite perfect though.

**Core**
 * Blueprints (currently in Sim)
    * [BeamBlueprint](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Sim/Blueprints/BeamBlueprintDefinition.lua)
    * [BlueprintBase](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Sim/Blueprints/BaseBlueprintDefinition.lua)
    * [EmitterBase](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Sim/Blueprints/EmitterBlueprintDefinition.lua)
    * [ProjectileBlueprint](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Sim/Blueprints/ProjectileBlueprintDefinition.lua)
    * [PropBlueprint](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Sim/Blueprints/PropBlueprintDefinition.lua)
    * [TrailEmitterBase](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Sim/Blueprints/TrailEmitterBlueprintDefinition.lua)
    * [UnitBlueprint](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Sim/Blueprints/UnitBlueprintDefinition.lua)
    * [WeaponBlueprint](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Sim/Blueprints/WeaponBlueprintDefinition.lua)
 * moho.EntityCategory[](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Core/EntityCategory.lua) ([pending creation](4041))
 * [moho.sound_methods](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Sim/CSound.lua) ([pending transfer from Sim](4041))

**User**
 * moho.discovery_service_methods[](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/User/CDiscoveryService.lua) ([pending creation](4024))

Functions
-----
The engine documentation for functions is all right. They're generally in the right `/engine/Sim.lua`, `/engine/User.lua`, or `/engine/Core.lua` file.

[**Core**](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Core.lua)
 * `AudioSetLanguage` (unmerged)
 * `EntityCategoryContains` (unmerged)
 * `EntityCategoryFilterDown` (unmerged)
 * `GetBlueprint` (unmerged)
 * `GetDepositsAroundPoint` (unmerged)
 * `GetGameTimeSeconds` (unmerged)
 * `GetTimeForProfile` (missing)
 * `GetUnitById` (unmerged)
 * `HasLocalizedVO` (unmerged)
 * `IsAlly` (unmerged)
 * `IsEnemy` (unmerged)
 * `IsNeutral` (unmerged)
 * `LuaDumpBinary` (missing)
 * `ParseEntityCategory` (unmerged)
 * `Random` (unmerged)
 * `SessionIsReplay` (unmerged)
 * `SetFocusArmy` (unmerged)
 * `_ALERT` (missing)
 * `_TRACEBACK` (missing)

[**User**](https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/User.lua)
 * `SoundIsPrepared` ([pending fixing](4024))
 * `StartSound` ([pending fixing](4024))
 * `StopAllSounds` ([pending fixing](4024))
 * `StopSound` ([pending fixing](4024))
 * `SyncPlayableRect` ([pending fixing](4024))
 * `TeamColorMode` ([pending fixing](4024))
 * `ToggleFireState` ([pending fixing](4024))
 * `ToggleScriptBit` ([pending fixing](4024))
 * `UISelectAndZoomTo` ([pending fixing](4024))
 * `UISelectionByCategory` ([pending fixing](4024))
 * `UIZoomTo` ([pending fixing](4024))
 * `UnProject` ([pending fixing](4024))
 * `ValidateIPAddress` ([pending fixing](4024))
 * `ValidateUnitsList` ([pending fixing](4024))
 * `WorldIsLoading` ([pending fixing](4024))
 * `WorldIsPlaying` ([pending fixing](4024))
 * `_c_CreateCursor` ([pending fixing](4024))
 * `_c_CreateDecal` ([pending fixing](4024))
 * `_c_CreatePathDebugger` ([pending fixing](4024))

**Variables**
-----
No documentation at all, which is fine since there's very little here anyway. It's also a little weird since it depends when the Lua state is initialized.

**Sim**
 * `__language` (missing)

**Core**
 * `__factory_objects` (missing)
 * `_LOADED` (missing)

**Core at Launch**
 * `__blueprints` (missing)
 * `__factory_objects` (missing)
 * `_LOADED` (missing)
 *  `categories` (missing)

**Comment by Garanas** [comment]
And the values as part of the [import](https://github.com/FAForever/fa/blob/deploy/fafdevelop/lua/system/import.lua#L7) mechanic, including `__diskwatch`.

And once you're in game, `__EngineStats` as populated by the engine.

**Comment by Hdt80bro** [comment]
Would we like to document items that the engine interacts with, but does not define? An example would be https://github.com/Hdt80bro/fa/blob/annotate-engine/user.lua/engine/User/CMauiEdit.lua#L158-L191

**Comment by Garanas** [comment]
Yes - that is very useful! There are a lot of strange Lua / engine interactions with unit functions

**Comment by Garanas** [comment]
Can this issue get converted to a Wiki entry, on this repository?

**Comment by Hdt80bro** [comment]
A Wiki entry? I'd prefer to just fix the documentation differences.
If you're talking about engine interactions though, then yes! That'd be a good idea.
